fly/types/bit_stream/bit_stream_reader.h

```diff
@@ -216,13 +217,25 @@ byte_type BitStreamReader::fill(DataType &buffer, byte_type bytes) noexcept
         detail::BitStreamTraits::is_unsigned_integer_v<DataType>,
         "DataType must be an unsigned integer type");
 
-    m_stream.read(
-        reinterpret_cast<std::ios::char_type *>(&buffer),
-        static_cast<std::streamsize>(bytes));
+    const std::ios::pos_type start = m_stream.tellg();
+    m_stream.seekg(0, std::ios::end);
 
-    buffer = endian_swap<Endian::Big>(buffer);
+    const std::ios::pos_type length = m_stream.tellg() - start;
+    m_stream.seekg(start, std::ios::beg);
 
-    return static_cast<byte_type>(m_stream.gcount());
+    byte_type bytesRead = 0;
+
+    if (m_stream.read(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/manual/trflynn89/libfly/pullRequest?prid=4795547)